alfresco-webscripts-client/alfresco-webscripts-ditto/src/main/java/eu/xenit/alfresco/webscripst/client/ditto/SlingShotFakeClient.java

```diff
@@ -0,0 +1,157 @@
+            metadata.setAssocs(toAssociations(Type.TARGET, node.getTargetAssociationCollection()));
+            metadata.setSourceAssocs(toAssociations(Type.SOURCE, node.getSourceAssociationCollection()));
+            metadata.setPermissions(null); // TODO not yet in ditto
+            return metadata;
+        }).orElse(null);
+    }
+
+    private NameContainer toNameContainer(QName qName) {
+        return new NameContainer(qName.toString(), qName.toPrefixString());
+    }
+
+    private List<NameContainer> toNameContainers(Set<QName> qnames) {
+        return qnames.stream()
+                .map(this::toNameContainer)
+                .collect(Collectors.toList());
+    }
+
+    private List<Property> toProperties(NodeProperties nodeProperties) {
+        List<Property> ret = new ArrayList<>();
+        nodeProperties.forEach((qName, serializable) -> {
+            ModelInfo info = modelHelper.getModelInfoByQName(qName);
+            NameContainer name = toNameContainer(qName);
+            NameContainer type = new NameContainer(info.getType().toString(), info.getType().toPrefixString());
+            String value = info.getDeserializer().apply(serializable);
+            ValueContainer valueContainer = new ValueContainer(info.getDataType(), value, info.isContent(),
+                    info.isNodeRef(), StringUtils.nullOrEmpty(value));
+            boolean multiple = isMultiple(serializable);
+            boolean residual = info.isResidual();
+            ret.add(new Property(name, valueContainer, type, multiple, residual));
+        });
+        return ret;
+    }
+
+    private List<Parent> toParentsFromChildCollection(ParentChildNodeCollection childAssocs) {
+        return toParents(childAssocs, true);
+    }
+
+    private List<Parent> toParentsFromParentCollection(ParentChildNodeCollection parentAssocs) {
+        return toParents(parentAssocs, false);
+    }
+
+    private List<Parent> toParents(ParentChildNodeCollection collection, boolean child) {
+        return collection.getAssociations()
+                .map(parentChildAssoc -> {
+                    Parent parent = new Parent();
+                    Node childNode = parentChildAssoc.getChild();
+                    Node parentNode = parentChildAssoc.getParent();
+
+                    parent.setName(new NameContainer(childNode.getName(), null));
+                    NodeReference nodeRef = child ? childNode.getNodeRef() : parentNode.getNodeRef();
+                    parent.setNodeRef(nodeRef.toString());
+                    QName type = child ? childNode.getType() : parentNode.getType();
+                    parent.setType(toNameContainer(type));
+
+                    QName assocTypeQName = parentChildAssoc.getAssocTypeQName();
+                    parent.setAssocType(toNameContainer(assocTypeQName));
+
+                    parent.setPrimary(parentChildAssoc.isPrimary());
+                    parent.setIndex(-1); // TODO how to determine
```

Comment:
TODO left on purpose?

Reply:
Yes, no idea how to get this information at the moment
I've only seen it contain -1 for now, but we shouldn't assume that this will always be the case